tests/unit-tests/test_trainer.py

```diff
@@ -1177,6 +1177,107 @@ def test_hgt_nc4ef():
+            fanout=config.fanout,
+            batch_size=config.batch_size,
+            label_field=config.label_field,
+            node_feats=config.node_feat_name,
+            edge_feats=config.edge_feat_name,
+            train_task=False)
+
+        evaluator4 = GSgnnClassificationEvaluator(config.eval_frequency,
+                                                    config.eval_metric,
+                                                    config.multilabel,
+                                                    config.use_early_stop)
+
+        trainer4.setup_evaluator(evaluator4)
+
+        trainer4.fit(
+            train_loader=train_dataloader4,
+            val_loader=val_dataloader4,
+            num_epochs=2
+            )
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+        gdata = GSgnnData(part_config=part_config)
+
+        # Test case 4: abnormal case, set HGT model with edge features for NC, but not
+        #              provide edge features.
+        #              This will trigger an assertion error, asking for giving edge feature
+        #              for message passing computation.
+        create_config4ef(Path(tmpdirname), 'gnn_nc.yaml', encoder='hgt', use_ef=True)
+        args = Namespace(yaml_config_file=os.path.join(Path(tmpdirname), 'gnn_nc.yaml'),
+                            local_rank=0)
+        config = GSConfig(args)
+
+        model5 = create_builtin_node_gnn_model(gdata.g, config, True)
+        trainer5 = GSgnnNodePredictionTrainer(model5)
+        trainer5.setup_device(device)
+
+        train_dataloader5 = GSgnnNodeDataLoader(
+            gdata,
+            target_idx=gdata.get_node_train_set(config.target_ntype),
+            fanout=config.fanout,
+            batch_size=config.batch_size,
+            label_field=config.label_field,
+            node_feats=config.node_feat_name,
+            edge_feats=None,
+            train_task=True)
+        val_dataloader5 = GSgnnNodeDataLoader(
+            gdata,
+            target_idx=gdata.get_node_val_set(config.target_ntype),
+            fanout=config.fanout,
+            batch_size=config.batch_size,
+            label_field=config.label_field,
+            node_feats=config.node_feat_name,
+            edge_feats=None,
+            train_task=False)
+
+        evaluator5 = GSgnnClassificationEvaluator(config.eval_frequency,
+                                                    config.eval_metric,
+                                                    config.multilabel,
+                                                    config.use_early_stop)
+
+        trainer5.setup_evaluator(evaluator5)
+
+        with assert_raises(AssertionError):
+            trainer5.fit(
+                train_loader=train_dataloader5,
+                val_loader=val_dataloader5,
+                num_epochs=2
+                )
+    
```

Comment:
Could you add another abnormal case, where set HGT model without edge features for NC, but provide edge features.

Reply:
I see there is already such test case. So ignore this comment.